interfaces/builtin/desktop.go

```diff
@@ -63,6 +68,57 @@ owner @{HOME}/.local/share/fonts/{,**} r,
 # some applications are known to mmap fonts
 /usr/{,local/}share/fonts/** m,
 
+# Allow access to xdg-document-portal file system.  Access control is
+# handled by bind mounting a snap-specific sub-tree to this location
+# (ie, this is /run/user/<uid>/doc/by-app/snap.@{SNAP_INSTANCE_NAME}
+# on the host).
+owner /run/user/[0-9]*/doc/{,*/} r,
+# Allow rw access without owner match to the documents themselves since
+# the user guided the access and can specify anything DAC allows.
+/run/user/[0-9]*/doc/*/** rw,
+
+# Allow access to xdg-desktop-portal and xdg-document-portal
+dbus (receive, send)
+    bus=session
+    interface=org.freedesktop.portal.*
+    path=/org/freedesktop/portal/{desktop,documents}{,/**}
+    peer=(label=unconfined),
+
+dbus (receive, send)
+    bus=session
+    interface=org.freedesktop.DBus.Properties
+    path=/org/freedesktop/portal/{desktop,documents}{,/**}
+    peer=(label=unconfined),
+
+# The portals service is normally running and newer versions of
+# xdg-desktop-portal include AssumedAppArmor=unconfined. Since older
+# systems don't have this and because gtkfilechoosernativeportal.c relies on
+# service activation, allow sends to peer=(name=org.freedesktop.portal.{Desktop,Documents})
+# for service activation.
+dbus (send)
+    bus=session
+    interface=org.freedesktop.portal.*
+    path=/org/freedesktop/portal/desktop{,/**}
+    peer=(name=org.freedesktop.portal.Desktop),
+dbus (send)
+    bus=session
+    interface=org.freedesktop.DBus.Properties
+    path=/org/freedesktop/portal/desktop{,/**}
+    peer=(name=org.freedesktop.portal.Desktop),
+dbus (send)
+    bus=session
+    interface=org.freedesktop.portal.*
+    path=/org/freedesktop/portal/documents{,/**}
+    peer=(name=org.freedesktop.portal.Documents),
+dbus (send)
+    bus=session
+    interface=org.freedesktop.DBus.Properties
+    path=/org/freedesktop/portal/documents{,/**}
+    peer=(name=org.freedesktop.portal.Documents),
```

Comment:
Unless my eyes are betraying me, the last two rules are identical to the first two.

Reply:
Your eyes are betraying you - the first 2 are Desktop the second two are Documents

Reply:
Ah, I see, indeed! :-)

My last question then: if I understand correctly, these four rules are essentially the same as the two above having `label=unconfined`, and are needed to handle the case where the services are not running and need to be activated by their service name. I would make this more obvious by adding a sentence at the beginning of the comment block above, saying something like: "Replicate the rules above specifying the well-known service name this time; this is needed for those cases when the services need to be DBus-activated."

Reply:
For what it is worth, these rules along with their comments are just being moved within the file (to separate core+classic and classic-only cases), rather than being new rules. I'd prefer not to go through rewording things that are essentially unmodified.